pkg/packagekit/autoupdates.jsx

```diff
@@ -203,16 +209,15 @@ class DnfImpl extends ImplBase {
 
 // Returns a promise for instantiating "backend"; this will never fail, if
 // automatic updates are not supported, backend will be null.
-function getBackend() {
-    if (!getBackend.promise) {
-        debug("getBackend() called first time, initializing promise");
+function getBackend(clearPromise) {
+    if (!getBackend.promise || clearPromise) {
+        debug("getBackend() called first time or clearPromise passed, initializing promise");
         let dfd = cockpit.defer();
         getBackend.promise = dfd.promise();
 
-        // TODO: only check for dnf/yum/apt and install the "automatic" packages dynamically
-        cockpit.script(["if rpm -q dnf-automatic >/dev/null; then echo dnf; " +
-                        "elif rpm -q yum-cron >/dev/null; then echo yum; " +
-                        "elif dpkg -s unattended-upgrades >/dev/null 2>&1; then echo apt; fi"],
+        cockpit.script(["(command -v dnf >/dev/null && echo dnf) || " +
+                        "(command -v yum >/dev/null && echo yum) || " +
+                        "(command -v apt >/dev/null && echo apt)"],
```

Comment:
Completely nerdy suggestion: `command -v dnf yum apt  | head -n1 | xargs basename`, or do the head/basename in JS.

Reply:
Haha ok, that's a lot more onelinery